[14.0][ADD] stock_picking_assigned, *_secondary_uom, *_batch_link

ChristianSantamaria opened the pull request:
Do not review, it's still in progress. 
The name of the addon will not be the final one

dalonsod commented on stock_picking_assigned_batch_link/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "version": "14.0.1.0.0",
+    "category": "Stock",
+    "website": "",
+    "depends": [
```
Dependency on `web_domain_filter` is missing, isn't it?

ChristianSantamaria committed:
[ADD] stock_picking_assigned

ChristianSantamaria committed:
[ADD] stock_picking_assigned_secondary_uom

ChristianSantamaria committed:
[ADD] stock_picking_assigned_batch_link

ChristianSantamaria committed:
[IMP] stock_picking_assigned: Some improvements 01/25
- Without printing and all ubication
- New report distribution ticket
- Origin location
- Mode assigned in top wizard
- Readonly variable mode
- Wrong reset lines wizard
- New field recounted to know when it has already been counted

ChristianSantamaria committed:
[IMP] stock_picking_assigned_secondary_uom: Some improvements 01/25
- New report distribution ticket
- Readonly variable mode
- Fix calculate secondary uom

ChristianSantamaria committed:
[IMP] stock_picking_assigned_batch_link: Some improvements 01/25
- Change batch clean product_id
- Readonly variable mode
- Fix Assign and Print not refresh wizard fields

ChristianSantamaria commented:
BIG SQUASH, created a backup in case something has been left behind

ChristianSantamaria committed:
[IMP] stock_picking_assigned_secondary_uom: validate no secondary unit in POL

ChristianSantamaria committed:
[IMP] stock_picking_assigned: exclude 0kg in lot field
